Artesian/_Query/BidAskQuery.py

```diff
@@ -3,55 +3,131 @@
+        """ Gets the BidAsk Query in a relative period range time window.
+        
+            Args:
+                pStart: the relative period start of the range of extracted timeseries. (ex. "P--3D")
+                pEnd: the relative period end of the range of the extracted timeseries. (ex. "P10D") 
+        """
+
         super()._inRelativePeriodRange(pStart, pEnd)
         return self
-    def inRelativePeriod(self, extractionPeriod):
+    def inRelativePeriod(self, extractionPeriod: str) -> _BidAskQuery:
+        """ Gets the BidAsk Query in a relative period of a time window.
+        
+            Args:
+                extractionPeriod: the relative period of extracted timeseries. (ex. "P5D")
+        """
         super()._inRelativePeriod(extractionPeriod)
         return self
-    def inRelativeInterval(self, relativeInterval):
+    def inRelativeInterval(self, relativeInterval: str) -> _BidAskQuery:
+        """ Gets the Relative Interval considers a specific interval of time window.
+        
+            Args:
+                relativeInterval: the relative interval of extracted timeseries. (ex. "RelativeInterval.ROLLING_WEEK" or "RelativeInterval.ROLLING_MONTH") 
+        """
         super()._inRelativeInterval(relativeInterval)
         return self
-    def forProducts(self, products):
+    def forProducts(self, products: str) -> _BidAskQuery:
+        """ Gets the Products tor the BidAsk Query in a time window.
+        
+            Args:
+                forProducts: ["D+1","Feb-18"]"""
         self._queryParameters.products = products
         return self
-    def withFillNull(self):
-        self._queryParameters.fill = NullFillStategy()
+    def withFillNull(self) -> _BidAskQuery:
+        """ Optional filler strategy for the extraction.
+        
+            Args: 
+                ex.  withFillNull() """
+        self._queryParameters.fill = _NullFillStategy()
         return self
-    def withFillNone(self):
-        self._queryParameters.fill = NoFillStategy()
+    def withFillNone(self) -> _BidAskQuery:
+        """ Optional filler strategy for the extraction.
+        
+            Args:
+                ex.  withFillNone() """
+        self._queryParameters.fill = _NoFillStategy()
         return self
-    def withFillLatestValue(self, period):
-        self._queryParameters.fill = FillLatestStategy(period)
+    def withFillLatestValue(self, period: str) -> _BidAskQuery:
+        """ Optional filler strategy for the extraction.
+        
+            Args:
+               period:  ex.  withFillLatestValue("P5D") """
+        self._queryParameters.fill = _FillLatestStategy(period)
         return self
-    def withFillCustomValue(self, **val):
-        self._queryParameters.fill = FillCustomStategy(val)
+    def withFillCustomValue(self, **val) -> _BidAskQuery:
+        """ Optional filler strategy for the extraction.
+        
+            Args:
```

Comment:
val parameter should be documented

example is .withFillCustomValue(bid=12, price=20) but I'm not sure if bid and price are correct.